### PR TITLE
Factor out dependency installation script

### PR DESCRIPTION
**Pull request by bjackman**
This provides an installation script that can be used to set up VMs and containers on Ubuntu-like systems. Then modifies the Vagrantfile to use that script.

I've kept this as 3 commits so that I can keep track of the changes from the script I was given by another team, but I'm happy to squash before merging.

### CODE REVIEW

File: install_base_ubuntu.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    # add more package names here, separated by space
+    pip_pkgs="Cython trappy bart-py devlib psutil wrapt ipython jupyter"
+
+
+    # check root permision
+    # No need to do this on LAVAv2 LXC container
+    # check_root
+
+    # remove some packages
+    for package in $remove_pkgs
```
Comment (derkling) [inline install_base_ubuntu.sh]: Are not these loops a bit overkilling?

Why not just a simple:

```bash
sudo apt-get -y remove $remove_pkgs
sudo apt-get -y install $pre_requisite_pkgs
```

The same for pip installed packages...

Comment (bjackman) [inline install_base_ubuntu.sh]: Agreed.

**Comment by bjackman** [comment]
Pushed a new version, I simplified it a bit but then also complicated it a bit...

**Comment by derkling** [comment]
Can we squash the first two patches together?

**Comment by bjackman** [comment]
OK, pushed a squashed version

**Comment by bjackman** [comment]
Uhh.. meant to only squash 2. Take two...

File: install_base_ubuntu.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+apt-get -y remove $remove_pkgs
+
+apt-get -y install $pre_requisite_pkgs
+
+yes w | pip install $pip_pkgs
+
+if [ "$install_android_sdk" == y ]; then
+    apt-get -y install openjdk-7-jre openjdk-7-jdk
+    ANDROID_SDK_URL="https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz"
+    if [ ! -e ./tools/android-sdk-linux ]; then
```
Comment (ionela-voinescu) [inline install_base_ubuntu.sh]: Using this as a relative path does not work for calls like: sudo ./lisa/install_base_ubuntu.sh.

**Comment by bjackman** [comment]
OK pushed a version with the `pip install --upgrade packaging appdirs`.

**Comment by bjackman** [comment]
:weary: 

New version with

- The `pip install --upgrade packaging appdirs` nonsense
- Got rid of the `--fast` thing and just set it to use the latest pip, this avoids compiling anything. But to make this work in the Vagrant provisioning environment where the `$PATH` is not suitable, we need to explicitly use `/usr/local/bin/pip`.

**Pull request merged**